pkg/ccl/kvccl/kvtenantccl/upgradeccl/tenant_upgrade_test.go

```diff
@@ -278,6 +277,8 @@ func TestTenantUpgradeFailure(t *testing.T) {
 									ctx context.Context, version clusterversion.ClusterVersion, deps upgrade.TenantDeps,
 								) error {
 									tenantStopperChannel <- struct{}{}
+									// Wait for a second here to give the tenant some time to be stopped.
```

Comment:
You could define a channel that gets `close()`d here, and wait on that below.

Reply:
We already wait on `waitForTenantClose` which only receives a value once the stopper returns (i.e. once all workers including the `v2` migration return).